src/NzbDrone.Core/Download/Clients/DownloadStation/Proxies/DSMInfoProxy.cs

```diff
@@ -13,20 +14,28 @@ public interface IDSMInfoProxy
 
     public class DSMInfoProxy : DiskStationProxyBase, IDSMInfoProxy
     {
-        public DSMInfoProxy(IHttpClient httpClient, Logger logger) :
-            base(httpClient, logger)
+        public DSMInfoProxy(IHttpClient httpClient, Logger logger, ICacheManager cacheManager) :
+            base(httpClient, logger, cacheManager)
         {
         }
 
+        protected override KeyValuePair<string, DiskStationApi> Api
+        {
+            get
+            {
+                return new KeyValuePair<string, DiskStationApi>("SYNO.DSM.Info", DiskStationApi.DSMInfo);
+            }
+        }
+
         public string GetSerialNumber(DownloadStationSettings settings)
         {
             var arguments = new Dictionary<string, object>() {
-                { "api", "SYNO.DSM.Info" },
-                { "version", "2" },
+                { "version", GetMinVersion(settings) },
```

Comment:
The risk here is that you know how to handle version 1 & 2 responses. What if you suddenly talk to a future DSM with version 3 that returns different data?